src/wilson/dom.cljs

```diff
@@ -174,9 +174,16 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (let [parse-f2c (fn [render-fn & outer-props]
+                    (let [outer-comp (apply render-fn outer-props)]
+                      (fn [_ & inner-props]
+                        (let [inner-comp (apply outer-comp inner-props)]
+                          (with-attrs {:class cls} inner-comp)))))]
+    (if (keyword? (first component))
```

Comment:
OK, cool :) You can write it either with or without match; match would have the benefit that you don't create parse-f2c when it's not necessary. You could also alternatively do that by making parse-f2c an external fn, and use `partial` to pass `cls` to it.